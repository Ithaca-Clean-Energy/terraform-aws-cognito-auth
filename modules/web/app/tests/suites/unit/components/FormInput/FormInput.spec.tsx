/*
 * Copyright (c) 2018 Martin Donath <martin.donath@squidfunk.com>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to
 * deal in the Software without restriction, including without limitation the
 * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
 * sell copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */

import { shallow } from "enzyme"
import * as React from "react"

import {
  FormInput,
  Render,
  RenderProps
} from "components/FormInput/FormInput"

/* ----------------------------------------------------------------------------
 * Tests
 * ------------------------------------------------------------------------- */

/* Form input component */
describe("components/FormInput", () => {

  /* Render component */
  describe("Render", () => {

    /* Default props */
    const props: RenderProps = {
      classes: {
        root: "__ROOT__"
      }
    }

    /* Test: should render with default props */
    it("should render with default props", () => {
      const wrapper = shallow(<Render {...props} />)
      expect(wrapper).toMatchSnapshot()
    })

    /* Test: should render with additional input props */
    it("should render with additional input props", () => {
      const wrapper = shallow(<Render {...props} type="password" />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  /* Enhanced component */
  describe("FormInput", () => {

    /* Test: should render with default props */
    it("should render with default props", () => {
      const wrapper = shallow(<FormInput />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
