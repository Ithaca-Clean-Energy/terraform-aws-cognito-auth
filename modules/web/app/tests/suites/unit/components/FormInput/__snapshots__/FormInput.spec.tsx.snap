// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FormInput FormInput should render correctly 1`] = `
<TextField
  InputLabelProps={
    Object {
      "required": false,
    }
  }
  autoCapitalize="false"
  autoCorrect="false"
  fullWidth={true}
  margin="dense"
  required={false}
  select={false}
  type="text"
/>
`;

exports[`components/FormInput Render should render with additional input props 1`] = `
<TextField
  InputLabelProps={
    Object {
      "required": false,
    }
  }
  autoCapitalize="false"
  autoCorrect="false"
  classes={
    Object {
      "root": "__ROOT__",
    }
  }
  fullWidth={true}
  margin="dense"
  required={false}
  select={false}
  type="password"
/>
`;

exports[`components/FormInput Render should render with default props 1`] = `
<TextField
  InputLabelProps={
    Object {
      "required": false,
    }
  }
  autoCapitalize="false"
  autoCorrect="false"
  classes={
    Object {
      "root": "__ROOT__",
    }
  }
  fullWidth={true}
  margin="dense"
  required={false}
  select={false}
  type="text"
/>
`;
